### PR TITLE
Gauge execution.rst

### PR DESCRIPTION
**Pull request by deeptikeshavaprasad**
Hi Zabil,
The following files are for review: writing-specifications.rst and execution.rst.

NOTE: I have worked on most of the content in both the files, except the following:
1) writing-specifications.rst : Content under ADVANCED and Troubleshooting
2) execution.rst : Content under Errors during execution and Troubleshooting

For now, I am working on the ADVANCED section offline. I didn't want to delay the review of the rest of the content in both the files because it's considerably huge.

It would be great if you could provide comments by Friday, July 19.

Thanks,
Deepti

### CODE REVIEW

File: writing-specifications.rst [hunk 1]
```diff
@@ -48,31 +63,51 @@ The following is an example of a Gauge specification:
...
     * Search for product "unknown"
     * The search results will be empty
 
+For more information about Markdown, see `Markdown Syntax <https://daringfireball.net/projects/markdown/syntax>`__.
 
-Specification Heading
-^^^^^^^^^^^^^^^^^^^^^
+Components of a specification
+-----------------------------
+A specification comprises of different sections; some of which are mandatory and few are optional.
```
Comment (zabil) [inline writing-specifications.rst]: `A specification comprises of different sections` might not be clear to all users. Can we replace `comprises` with a simpler word?

File: writing-specifications.rst [hunk 2]
```diff
@@ -196,29 +223,20 @@ The values written in *quotes*, ``"``, are parameters that are passed into the e
...
-Example
-^^^^^^^
-
-.. code-block:: gauge
-
-    * Login into my app
-    * Search for "gauge"
-    * Search for "gauge-java"
+| For more information about Context Steps and Tear Down steps, see :ref:`Context Steps<longstart-context>` and :ref:`longstart-teardown`.
+| For more informatin about how to write step implementations for different languages, see :ref:`language-steps`.
```
Comment (zabil) [inline writing-specifications.rst]: `informatin` spelling mistake here

File: writing-specifications.rst [hunk 3]
```diff
@@ -196,29 +223,20 @@ The values written in *quotes*, ``"``, are parameters that are passed into the e
...
 Steps are defined to take values as parameters so that they can be reused with different parameter values.
-The equivalent code of the step implementation for that particular language plugin must also have the same number of parameters as mentioned in the step.
 
-Example
-^^^^^^^
+The equivalent code of the step implementation for that particular language plugin must also have the same number of parameters as mentioned in the step.
 
+Examples
+^^^^^^^^
 In the following example, parameter values are ``“Die Hard”`` (within double quotes) within the two steps.
```
Comment (zabil) [inline writing-specifications.rst]: Change from `Die Hard` to `Cup Cakes`

File: writing-specifications.rst [hunk 4]
```diff
@@ -231,6 +249,7 @@ In the following example, parameter values are ``“Die Hard”`` (within double
     * Search for product "Die Hard"
     * "Die Hard" should show up in the search results
```
Comment (zabil) [inline writing-specifications.rst]: Change `Die Hard` to `Cup Cakes`

File: writing-specifications.rst [hunk 5]
```diff
@@ -244,15 +263,13 @@ A step can have the following types of parameters:
...
-1. Simple parameters
+Simple parameters
 ^^^^^^^^^^^^^^^^^^^^
 
 Simple parameters are values, which are used in a step within double quotes.
 
 Example
 ^^^^^^^
-
 In the following examples, ``Die Hard``, ``gauge-java``, and ``100`` are values within double quotes and are used in steps:
```
Comment (zabil) [inline writing-specifications.rst]: Change `Die Hard` to `Cup Cakes` Please make sure all occurrences of `Die Hard` is changed to `Cup Cakes`

File: writing-specifications.rst [hunk 6]
```diff
@@ -261,26 +278,18 @@ In the following examples, ``Die Hard``, ``gauge-java``, and ``100`` are values
...
 Dynamic parameters have the following syntax: ``<dynamic_param>``.
 
-Dynamic parameters are primarily used when referring to a table column value in :ref:`table_driven_execution`,
-or while passing values to :ref:`concept`.
-
 Example: Dynamic Parameters used in a spec which has a data table
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 In the following example, ``<name>`` is a dynamic parameter. 
-``<name>`` can take the values mentioned in the "name" column of the table. In this case, "Vishnu", "Prateek", or "Navaneeth".
+``<name>`` can take the values mentioned in the "name" column of the table. In this case, the values are "Vishnu", "Prateek", or "Navaneeth".
```
Comment (zabil) [inline writing-specifications.rst]: Please change the names to `Alice`, `Bob` and `Eve` Please make sure that all occurrences of Prateek, Vishnu and Navaneeth are changed to these values.

File: writing-specifications.rst [hunk 7]
```diff
@@ -451,38 +468,33 @@ The following is a sample CSV file, with the first row as table header and follo
...
-Tags can be defined in multiple lines
+    ## Successful search
+       Tags: successful
 
-Both the ``Login specification`` and the scenario
-``Successful login scenario`` have tags in multiple lines in the below example.
+Example: Multi-line tag
+^^^^^^^^^^^^^^^^^^^^^^^
+| In the following example, both the specification and scenario have tags in multiple lines.
+| Tags have to be indented while specifying in multiple lines.
```
Comment (zabil) [inline writing-specifications.rst]: `while specifying` can we change it to `when written` ?

File: writing-specifications.rst [hunk 8]
```diff
@@ -496,66 +508,85 @@ Both the ``Login specification`` and the scenario
...
--  Concepts are used inside a spec just like any other step. The
-   appropriate parameters are passed to them.
--  On execution all the steps under the concepts are executed in the
-   defined order.
+    For an existing product name, the search result will contain the product name
 
-**Note:** A single .cpt file can contain multiple concept definitions.
+    * User must be logged in as "admin"
+    * Open the product search page
+    * Search for product "Die Hard"
```
Comment (zabil) [inline writing-specifications.rst]: Please change from "Die Hard" to `Cup Cakes`

File: writing-specifications.rst [hunk 9]
```diff
@@ -496,66 +508,85 @@ Both the ``Login specification`` and the scenario
...
+    * "Die Hard" should show up in the search results
```
Comment (zabil) [inline writing-specifications.rst]: Please change from "Die Hard" to Cup Cakes

File: execution.rst [hunk 10]
```diff
@@ -1,291 +1,255 @@
...
-These scenarios can also belong to different specifications.
-
-To execute scenarios, ``gauge`` takes path to a specification file, 
-followed by a colon and the line number of the scenario. 
-Any line number which the scenario spans across can be used. 
-
-For example, in the above spec file, both the below 
-commands will run the same scenario.
+Example - multipe directories
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
[the remark this answers is not in the document]
Comment (deeptikeshavaprasad) [reply execution.rst]: Is there any way to perform a spell check from within the editor? I am aware that there's a prompt for choosing the words, but it helps if there's something that can be used on the entire file.

File: execution.rst [hunk 11]
```diff
@@ -1,291 +1,255 @@
...
+In the following example, specifications stored at ``specs`` and ``test_specs`` are run:
 
 .. code-block:: console
 
-    gauge run specs/helloworld.spec:3 # Runs scenario 'Admin Login'
-    gauge run specs/helloworld.spec:5 # Runs scenario 'Admin Login'
+    gauge run specs test_specs
 
-Consider a specification file, ``spec1.spec``:
+Example - sub-directories
```
[the remark this answers is not in the document]
Comment (deeptikeshavaprasad) [reply execution.rst]: So, basically, I have tried to maintain the following for consistency sake: document the syntax and then provide examples. Though at few instances, the "syntax" is obvious or simple, I have gone ahead just to be consistent. 
Hence, what you see at line 58 is just the syntax and later is actually an example. I have run the commands on my system and then included them as examples.

File: execution.rst [hunk 12]
```diff
@@ -343,214 +316,350 @@ Tags                               Selects specs/scenarios that
...
 
-Specs can be executed in parallel to run the tests faster and distribute
-the load.
+Based on the Tags expressions table, ``Successful Search`` scenario is run.
 
-This can be done by the command:
+Verbose reporting
+-----------------
+
+By default, Gauge generates reports at the specification level when executing tests.
```
[the remark this answers is not in the document]
Comment (deeptikeshavaprasad) [reply execution.rst]: I know I did mention that "execute" might not sound very appropriate, but in some instances, "execute" works fine. Especially, in a sentence when multiple instances of "execute" are present, replacing every instance with "run" really doesn't appear very reader-friendly. 
However, I think I'll use the word "run" as much as possible especially single instances of execute. I'll give it another look.

File: writing-specifications.rst (hunk 1 above)
Comment (deeptikeshavaprasad) [reply writing-specifications.rst]: How about:
A specification consists of different sections
OR
A specification contains different sections

File: execution.rst (hunk 12 above)
[the remark this answers is not in the document]
Comment (deeptikeshavaprasad) [reply execution.rst]: I have rephrased "verbose reporting", does this sound okay?

By default, Gauge generates reports that display whether a spec has passed or failed when executed. To ease debugging, reports can also be generated for every step to display whether a step has been executed successfully. You can use the ``--verbose`` flag to generate such reports.
These reports are generated on the console.

File: execution.rst (hunk 10 above)
[the remark this answers is not in the document]
Comment (zabil) [reply execution.rst]: You could try installing a plugin for visual studio code.

File: execution.rst (hunk 12 above)
[the remark this answers is not in the document]
Comment (zabil) [reply execution.rst]: Yes

File: writing-specifications.rst (hunk 1 above)
Comment (zabil) [reply writing-specifications.rst]: `A specification consists of different sections` works fine

File: execution.rst (hunk 10 above)
[the remark this answers is not in the document]
Comment (deeptikeshavaprasad) [reply execution.rst]: Done. Neha shared it with me.

**Comment by deeptikeshavaprasad** [comment]
Hi Zabil,
I have committed the changes and pushed the branch to origin.
Thanks,
Deepti

**Pull request merged**